airflow/jobs/scheduler_job.py

```diff
@@ -221,26 +221,31 @@ def is_alive(self, grace_multiplier: float | None = None) -> bool:
 
     def __get_concurrency_maps(
         self, states: list[TaskInstanceState], session: Session
-    ) -> tuple[DefaultDict[str, int], DefaultDict[tuple[str, str], int]]:
+    ) -> tuple[
+        DefaultDict[str, int], DefaultDict[tuple[str, str], int], DefaultDict[tuple[str, str, str], int]
+    ]:
```

Comment:
May be a good idea to give this type an alias (or may it a named tuple?)

Reply:
Also the fact that these are default dicts is actually not significant. Maybe it’s better to simplify it to `dict`.

Reply:
Since we get the concurrency for each run and task from the created maps, and since some of those runs and tasks are not present in the maps (default concurrency is 0), I prefer to keep it a default dict instead of using `.get(<id>, 0)` everywhere ([ex](https://github.com/apache/airflow/blob/main/airflow/jobs/scheduler_job.py#L431-L432)).

And for the method typing, I agree but I prefer `Dataclass` over `NamedTuple`, WDYT?